aiohttp/helpers.py

```diff
@@ -279,6 +329,246 @@ def basicauth_from_netrc(netrc_obj: Optional[netrc.netrc], host: str) -> BasicAu
     return BasicAuth(username, password)
 
 
+DigestFunctions: Dict[str, Callable[[bytes], "hashlib._Hash"]] = {
+    "MD5": hashlib.md5,
+    "MD5-SESS": hashlib.md5,
+    "SHA": hashlib.sha1,
+    "SHA256": hashlib.sha256,
+    "SHA512": hashlib.sha512,
+}
+
+
+class DigestAuthChallenge(TypedDict, total=False):
+    realm: str
+    nonce: str
+    qop: str
+    algorithm: str
+    opaque: str
+    ...
+
+
+class DigestAuthContext(threading.local):
```

Comment:
I'm a bit confused why we would need thread local storage since `aiohttp` is an asyncio library we shouldn't have multiple threads.